### PR TITLE
perf(core): use `ngDevMode` to tree-shake warning

### PR DESCRIPTION
**Pull request by arturovt**
## PR Checklist
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by atscott — APPROVED** [review]

**Review by jessicajaniuk — APPROVED** [review]
reviewed-for: size-tracking

File: packages/core/src/application_ref.ts [hunk 1]
```diff
@@ -734,8 +733,9 @@ export class ApplicationRef {
     });
 
     this._loadComponent(compRef);
-    if (isDevMode()) {
-      this._console.log(
+    if (typeof ngDevMode === 'undefined' || ngDevMode) {
+      const _console = this._injector.get(Console);
```
Comment (AndrewKushnir) [inline packages/core/src/application_ref.ts:737]: I believe we should retain `isDevMode()` call (to make things work in ViewEngine) here and just add the `ngDevMode` to make `Console` tree-shakable in Ivy:
```suggestion
    if ((typeof ngDevMode === 'undefined' || ngDevMode) && isDevMode()) {
      const _console = this._injector.get(Console);
```
The same pattern is used in the `application_ref.ts` file above, see:
https://github.com/angular/angular/blob/master/packages/core/src/application_ref.ts#L764-L770

**Review by AndrewKushnir — CHANGES_REQUESTED** [review]
@arturovt thanks for improving the code to make it more tree-shakable, great to see payload size limits to decrease 👍 

I've added a comment that we need to retain `isDevMode()` to make sure the code works correctly in ViewEngine.

Also I've applied the "breaking change" label, since this PR changes the contract of the class that is a part of the public API. I believe the impact would be minimal, but we should make sure to communicate it correctly (with the "BREAKING CHANGE" note in commit message).

**Comment by AndrewKushnir** [comment]
Quick update: @atscott pointed out that `ApplicaitonRef` class constructor has the `@internal` annotation, so constructor is not included into d.ts file, so it's a private API and we can change it without triggering the breaking changes procedure (so I've updated labels on this PR accordingly). Thank you.

**Comment by arturovt** [comment]
@AndrewKushnir I've updated the code. Not sure why Circle workflow is failing :man_shrugging:

**Comment by AndrewKushnir** [comment]
@arturovt thanks for additional changes. It looks like there is one more place that needs an update:
https://github.com/angular/angular/blob/12f7101d9213f59c45a3c377e7cfa246aa0c44a6/packages/core/src/application_module.ts#L78-L82
(the `Console` should be removed from the `deps` array and the `Console` symbol import should be dropped as well from that file)

**Comment by arturovt** [comment]
Thanks Andrew, that worked! I will re-check the payload size again after that change and will update it (if needed) if the next few hours.

**Review by AndrewKushnir — APPROVED** [review]
LGTM, thanks @arturovt 👍

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: size-tracking

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:357878827:BASE:357878838:1613543146373:25b1ec77).

**Comment by AndrewKushnir** [comment]
@arturovt FYI, internal tests in Google's codebase went well, so I'm adding this PR to the merge queue.

Thanks for contributing to Angular!

**Pull request closed without merging**